src/main/java/zaasclient/HttpsClient.java

```diff
@@ -0,0 +1,201 @@
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package zaasclient;
+
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.*;
+import java.security.cert.CertificateException;
+
+public class HttpsClient implements Closeable {
+    private CloseableHttpClient closeableHttpsClient;
+    private final String keyStorePassword;
+    private final String trustStorePassword;
+    private final String keyStoreType;
+    private final String trustStoreType;
+    private final RequestConfig requestConfig;
+    private String trustStorePath;
+    private String keyStorePath;
+
+    public HttpsClient(ConfigProperties configProperties) {
+        this.requestConfig = this.buildCustomRequestConfig();
+        this.keyStorePath = configProperties.getKeyStorePath();
+        this.keyStorePassword = configProperties.getKeyStorePassword();
+        this.keyStoreType = configProperties.getKeyStoreType();
+        this.trustStorePath = configProperties.getTrustStorePath();
+        this.trustStorePassword = configProperties.getTrustStorePassword();
+        this.trustStoreType = configProperties.getTrustStoreType();
+    }
+
+
+    public CloseableHttpClient getHttpsClientWithTrustStore()
+        throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException, KeyManagementException {
+        SSLContext sslContext;
+        TrustManagerFactory tmf = null;
+
+        if (trustStorePath != null) {
+            tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
+        sslContext = this.getSSLContext(null, tmf);
+
+        return getCloseableHttpClient(sslContext);
+    }
+
+    /**
+     * an overloaded version of getHttpsClientWithTrustStore for the query api
+     *
+     * @param cookieStore
+     * @return
+     * @throws NoSuchAlgorithmException
+     * @throws KeyStoreException
+     * @throws IOException
+     * @throws CertificateException
+     * @throws KeyManagementException
+     */
+    public CloseableHttpClient getHttpsClientWithTrustStore(BasicCookieStore cookieStore)
```

Comment:
The code block in lines 79-89 is the same as in the other class getHttpsClientWithTrustStore(). We can extract it to a separate private method such as getTrustManagerFactory()